packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +75,14 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    // TODO(eseidel): This clearly isn't correct.  Not sure what would be.
-    // Should we use the ios images on ios?
-    String density = 'drawable-xxhdpi';
+    String density = new BreakpointQualifier(
+      candidates: _kIconDensities,
+      value: MediaQuery.of(context).devicePixelRatio
+    ).resolve();
```

Comment:
The Qualifier abstraction made more sense in a previous iteration, I'll get rid of it.